composer/datasets/synthetic.py

```diff
@@ -21,6 +29,88 @@ class SyntheticDataLabelType(StringEnum):
     CLASSIFICATION_ONE_HOT = "classification_one_hot"
 
 
+class SyntheticBertTokenizer(BertTokenizer):
+
+    def __init__(self, dataset, vocab_size=256):
+        try:
+            import tokenizers
+        except ImportError as e:
+            raise ImportError(
+                'Composer was installed without NLP support. To use NLP with Composer, run: `pip install mosaicml[nlp]`.'
+            ) from e
+
+        tokenizer = tokenizers.Tokenizer(tokenizers.models.WordPiece())
+        tokenizer.enable_padding(direction="right", pad_id=0, pad_type_id=0, pad_token="[PAD]", pad_to_multiple_of=8)
+        tokenizer.normalizer = tokenizers.normalizers.NFKC()
+        tokenizer.pre_tokenizer = tokenizers.pre_tokenizers.ByteLevel()
+        tokenizer.decoder = tokenizers.decoders.ByteLevel()
+        trainer = tokenizers.trainers.WordPieceTrainer(
+            vocab_size=vocab_size,
+            initial_alphabet=tokenizers.pre_tokenizers.ByteLevel.alphabet(),
+            special_tokens=["[PAD]", "[UNK]", "[SEP]", "[CLS]", "[MASK]"],
+        )
+        tokenizer.train_from_iterator(dataset, trainer=trainer)
+        tmp_tokenizer_file = NamedTemporaryFile()
```

Comment:
Can you use a context manager for the temporary file, and then call `super().__init__` within this context manager, so the tempfile will automatically disappear once its no longer needed?